backend/logic/courses/courses.js

```diff
@@ -261,12 +261,102 @@ var updateCourse = async (course, newTitle, newTags) => {
   }
 };
 
+/**
+ * Assign a course to a curriculum
+ * @author Yufei Liu
+ * @param {string} curriculumType,{string} courseCode,{string} curriculum
+ * @returns true if successful
+ * @throws error if curriculumType, courseCode or curriculum dose not exist
+ *         course already assigned a curriculum
+ *
+ */
+var assignCourseToCurriculum = async (
+  curriculumType,
+  courseCode,
+  curriculum
+) => {
+  let conn = await mysql.getNewConnection();
+
+  const checkExistQuery = `SELECT *
```

Comment:
I may be wrong, but this prevents a course from being assigned to multiple curriculum.  For example, if a course_A is assigned to curriculum_A, it will be returned and you're no longer able to assign it to curriculum_B

Reply:
Ohhh you are right, I forget to  check curriculum. It should check duplicate for both course_code and curriculum.

Reply:
I agree I think you should check if there are duplicate course_codes within the given curriculum only :)

Reply:
This would be good for now. For the future, we can add a composite key on course_code and curriculum to prevent insertion